portfolio/src/main/webapp/script.js

```diff
@@ -37,4 +37,35 @@ function age() {
     document.getElementById("age").innerText = `${age} years old`;
 }
 
-// TODO: create a function that returns the date in MM/DD/YYY format
\ No newline at end of file
+// TODO: create a function that returns the date in MM/DD/YYY format
+
+function getText() {
+    age() // you can only have one function in the onload attribute
+    console.log("Fetching text.");
+    const responsePromise = fetch("/data");
+    responsePromise.then(handleResponse);
+}
+
+function handleResponse(response) {
+    console.log("Handling the response.");
+    const textPromise = response.text();
+    textPromise.then(addTexttoDOM);
+}
+
+function addTexttoDOM(text) {
```

Comment:
This should probably be camel case. It is a bit weird because of DOM so you might want to double check what to do thin this case, but I think addTextToDom might be better.

Reply:
fixed!